sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2059,7 +2061,8 @@ pi_result piContextRelease(pi_context Context) {
     // and therefore it must be valid at that point.
     // Technically it should be placed to the destructor of pi_context
     // but this makes API error handling more complex.
-    ZE_CALL(zeContextDestroy(ZeContext));
+    if (DestoryZeContext)
```

Comment:
NIT: I am not sure why introducing a new variable "DestroyZeContext" is better than just directly checking "Context->OwnZeContext" in this if-condition.

Reply:
because "Context" is deleted at  this point already

Reply:
Hmmm... 
Context is still alive after Context->finalize(). 
Only command-list and eventpool that are associated with the context is destroyed.

Reply:
see line 2058

Reply:
Oops. I missed that. Thanks.